examples/log.c

```diff
@@ -337,17 +338,57 @@ static void print_time(const git_time *intime, const char *prefix)
 }
 
 /** Helper to print a commit object. */
-static void print_commit(git_commit *commit)
+static void print_commit(git_commit *commit, struct log_options options)
 {
 	char buf[GIT_OID_HEXSZ + 1];
 	int i, count;
 	const git_signature *sig;
 	const char *scan, *eol;
+	unsigned long length = 0;
 
 	git_oid_tostr(buf, sizeof(buf), git_commit_id(commit));
 	printf("commit %s\n", buf);
 
-	if ((count = (int)git_commit_parentcount(commit)) > 1) {
+	sig = git_commit_author(commit);
+	scan = git_commit_message(commit);
+	count = (int)git_commit_parentcount(commit);
+
+	if (options.show_log_size) {
+
```

Comment:
You're using newlines quite excessively. It makes sense to group logical blocks together by whitespaces, but we usually avoid using empty lines directly after a block-opening brace. Furthermore, we normally don't use two successive empty lines